test/CMakeLists.txt

```
@@ -85,119 +86,177 @@ list(APPEND TESTS connect_dend)
 # Add pytest
 # =============================================================================
 if(NRN_ENABLE_PYTHON AND PYTEST_FOUND)
-  add_test(NAME basic_tests COMMAND ${PYTHON_EXECUTABLE} -m pytest --cov-report=xml --cov=neuron --cov-append ${PROJECT_SOURCE_DIR}/test/pynrn
-           WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test/pynrn)
+  add_test(
+    NAME basic_tests
+    COMMAND
+      ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.basic_tests ${PYTHON_EXECUTABLE} -m pytest
+      --cov-report=xml --cov=neuron ${PROJECT_SOURCE_DIR}/test/pynrn
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test/pynrn)
   list(APPEND TESTS basic_tests)
   add_custom_target(
     pynrn_mod ALL
-    COMMAND
-      ${CMAKE_COMMAND} -E env ${TEST_ENV} $ENV{SHELL} ${CMAKE_BINARY_DIR}/bin/nrnivmodl
+    COMMAND ${CMAKE_COMMAND} -E env ${TEST_ENV} ${CMAKE_BINARY_DIR}/bin/nrnivmodl
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test/pynrn
     DEPENDS ${PROJECT_SOURCE_DIR}/test/pynrn/unitstest.mod nocmodl)
   add_dependencies(pynrn_mod nrniv)
 
   add_custom_target(
     rxdmod ALL
-    COMMAND
-      ${CMAKE_COMMAND} -E env ${TEST_ENV} $ENV{SHELL} ${CMAKE_BINARY_DIR}/bin/nrnivmodl ./ecs
+    COMMAND ${CMAKE_COMMAND} -E env ${TEST_ENV} ${CMAKE_BINARY_DIR}/bin/nrnivmodl ./ecs
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test/rxd)
   add_dependencies(rxdmod nrniv)
 
   if(NRN_ENABLE_RX3D)
-    add_test(NAME rxd_tests COMMAND ${PYTHON_EXECUTABLE} -m pytest --cov-report=xml --cov=neuron --cov-append ${PROJECT_SOURCE_DIR}/test/rxd)
+    add_test(
+      NAME rxd_tests
+      COMMAND
+        ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.rxd_tests ${PYTHON_EXECUTABLE} -m pytest
+        --cov-report=xml --cov=neuron ${PROJECT_SOURCE_DIR}/test/rxd)
     list(APPEND TESTS rxd_tests)
     if(NRN_ENABLE_MPI)
       find_python_module(mpi4py)
       if(mpi4py_FOUND)
         add_test(
           NAME rxd_mpi_tests
           COMMAND
-            ${MPIEXEC} -np 1 ${PYTHON_EXECUTABLE} -m pytest  --cov-report=xml --cov=neuron --cov-append ${PROJECT_SOURCE_DIR}/test/rxd --mpi)
+            ${MPIEXEC} -np 1 ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.rxd_mpi_tests
```

Would this work :
```suggestion
            ${MPIEXEC} -np 1 ${CMAKE_COMMAND} -E env COVERAGE_FILE=.coverage.${NAME}
```
I guess not

---

It doesn't work, unfortunately.